src/plugins/csvstorage/testmod_csvstorage.c

```diff
@@ -84,6 +84,21 @@ static void testwriteinvalidheader (const char * file)
 	PLUGIN_CLOSE ();
 }
 
+static void testwriteInvalidMetaMustFail (const char * file)
+{
+	Key * parentKey = keyNew ("user:/tests/csvstorage", KEY_VALUE, srcdir_file (file), KEY_END);
+	KeySet * conf = ksNew (20, keyNew ("system:/delimiter", KEY_VALUE, ";", KEY_END),
+			       keyNew ("system:/header", KEY_VALUE, "colname", KEY_END), KS_END);
+	KeySet * ks = ksNew (1, keyNew ("user:/tests/csvstorage", KEY_VALUE, "asdf", KEY_META, "asdf", "asdf", KEY_END), KS_END);
+	PLUGIN_OPEN ("csvstorage");
+	succeed_if (plugin->kdbSet (plugin, ks, parentKey) == ELEKTRA_PLUGIN_STATUS_ERROR,
+		    "kdbSet did not error on invalid meta key insertion");
+	ksDel (conf);
```

Comment:
Sorry, your solution was correct!

You used the macro `PLUGIN_OPEN(NAME)` which calls the function `elektraPluginOpen(...)` with the KeySet `conf` as argument.
https://github.com/ElektraInitiative/libelektra/blob/6a6d1bf57cc50348cbfbe9c94da7dc39c67969ad/tests/cframework/tests_plugin.h#L14-L22

There, the KeySet is assigned to an internal variable:
https://github.com/ElektraInitiative/libelektra/blob/6a6d1bf57cc50348cbfbe9c94da7dc39c67969ad/src/libs/elektra/plugin.c#L92

At the end of your test, the `PLUGIN_CLOSE()` macro is called and there `ksDel (handle->conf)` is done:

https://github.com/ElektraInitiative/libelektra/blob/6a6d1bf57cc50348cbfbe9c94da7dc39c67969ad/src/libs/elektra/plugin.c#L135


```suggestion
```